### PR TITLE
Before dev

### PR DESCRIPTION
**Pull request by DilsherB**
In this PR i did the following:

## Highlights
1. Poly-Morphysim used
2. Getters and Setters used
3. Initializers(constructor) used
4. private methods for each class used
5. In `person.rb` file created initializer and two methods
6. In `student.rb` file created initializer and one method
7. In `teacher.rb` file created initializer and modified one method.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require_relative 'Person'
+
+class Student < Person
+  attr_accessor :class_room
+
+  def initialize(class_room)
+    super
+    @class_room = class_room
+  end
```
Comment (Whoistolu) [inline student.rb]: - [x] [REQUIRED] Kindly ensure that the Student class's initialize method has the same parameters as the Person class's initialize method, in addition to the classroom parameter. This means including age, name, and parent_permission as parameters.
Also:
- [x] [REQUIRED] Pass all these parameters to the super call inside the initialize method of the Student class. This is important to properly initialize the attributes of the parent class.

With these changes, the Student class will be able to initialize properly by accepting all the necessary parameters from the Person class and additionally taking the classroom parameter. The super call within the initialize method will correctly pass these parameters to the constructor of the parent class (Person).

Here is what your code should look like afterward...

``` ruby
class Student < Person
 # maintain the former attr
  def initialize(age, classroom, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @classroom = classroom
  end
  # maintain former play_hooky method
end
```

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+require_relative 'Person'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(specialization)
+    super
+    @specialization = specialization
+  end
```
Comment (Whoistolu) [inline teacher.rb]: - [x] [REQUIRED] Let's do the same here. Kindly pass the age, name, and parent_permission into the initialize constructor and then use these parameters in the super.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @DilsherB 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### Highlight
🔆 There is an individual `.rb` file for each class (`Person`, `Student`, and `Teacher` )

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers, and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (DilsherB) [reply student.rb]: changes applied.

File: teacher.rb (hunk 2 above)
Comment (DilsherB) [reply teacher.rb]: changes applied.

**Review by ShadyShawkat — APPROVED** [review]
## Hi @DilsherB, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**